src/scenes/Event/scenes/oracle.js

```diff
@@ -589,74 +538,77 @@ export default class OraclePage extends React.Component {
         warningTypeClass: EventWarningType.Info,
       };
     }
 
-    // Did not select a result
-    if (!(token === Token.Bot && status === OracleStatus.WaitResult) && currentOptionIdx === -1) {
-      return {
-        disabled: true,
-        message: <FormattedMessage
-          id="oracle.selectResultDisabledText"
-          defaultMessage="Please click and select one of the options."
-        />,
-        warningTypeClass: EventWarningType.Info,
-      };
-    }
-
-    // Did not enter an amount
-    if (status === OracleStatus.Voting && (voteAmount <= 0 || Number.isNaN(voteAmount))) {
+    // Trying to set result or vote when not enough QTUM or BOT
+    const totalQtum = _.sumBy(walletAddresses, (wallet) => wallet.qtum ? wallet.qtum : 0);
+    const filteredAddress = _.filter(walletAddresses, { address: lastUsedAddress });
+    const totalBot = filteredAddress.length > 0 ? filteredAddress[0].bot : 0;
+    if ((token === Token.Bot || (token === Token.Qtum && status === OracleStatus.OpenResultSet))
+      && (totalQtum < maxTransactionFee && totalBot < oracle.consensusThreshold)
+      && [OracleStatus.Voting, OracleStatus.WaitResult, OracleStatus.OpenResultSet].includes(status)) {
       return {
         disabled: true,
-        message: <FormattedMessage
-          id="oracle.enterAmountDisabledText"
-          defaultMessage="Please entered a valid amount."
-        />,
-        warningTypeClass: EventWarningType.Info,
+        id: 'str.notEnoughQtumAndBot',
+        message: 'You don\'t have enough QTUM or BOT',
+        warningTypeClass: EventWarningType.Error,
       };
     }
 
     // Trying to bet more qtum than you have
-    const totalQtum = _.sumBy(walletAddresses, (wallet) => wallet.qtum ? wallet.qtum : 0);
-    if (token === Token.Qtum && status === OracleStatus.Voting && voteAmount > totalQtum) {
+    if ((status === OracleStatus.Voting && voteAmount > totalQtum + maxTransactionFee)
+      || totalQtum < maxTransactionFee) {
       return {
         disabled: true,
-        message: <FormattedMessage
-          id="str.notEnoughQtum"
-          defaultMessage="You don't have enough QTUM"
-        />,
+        id: 'str.notEnoughQtum',
+        message: 'You do\'t have enough QTUM',
         warningTypeClass: EventWarningType.Error,
       };
     }
 
     // Not enough bot for setting the result or voting
-    const filteredAddress = _.filter(walletAddresses, { address: lastUsedAddress });
-    const currentBot = filteredAddress.length > 0 ? filteredAddress[0].bot : 0;
     if ((token === Token.Qtum
       && (status === OracleStatus.WaitResult || status === OracleStatus.OpenResultSet)
-      && currentBot < oracle.consensusThreshold)
-      || (token === Token.Bot && status === OracleStatus.Voting && currentBot < voteAmount)) {
+      && totalBot < oracle.consensusThreshold)
```

Comment:
Revert. BOT & QTUM works differently. You can spend qtum from different addresses for the same tx but not BOT. That's why you are not supposed to compare with totalBot.